src/packet/public_key_encrypted_session_key.js

```diff
@@ -128,6 +128,12 @@ PublicKeyEncryptedSessionKey.prototype.encrypt = async function (key) {
  */
 PublicKeyEncryptedSessionKey.prototype.decrypt = async function (key) {
   const algo = enums.write(enums.publicKey, this.publicKeyAlgorithm);
+  const expectedAlgo = enums.write(enums.publicKey, key.algorithm);
+  // check that session key algo matches the secret key algo
+  // considering rsa_encrypt and rsa_encrypt_sign as equal
+  if ((algo > 2 || expectedAlgo > 2) && algo !== expectedAlgo) {
```

Comment:
I don't think you have to consider rsa_encrypt and rsa_encrypt_sign as equal. ProtonMail/crypto has this: https://github.com/ProtonMail/crypto/blob/1fa7f403fb9c3aa7b47443e1e3b39c80d6eedcb6/openpgp/packet/encrypted_key.go#L91-L93 and it hasn't caused any issues that I know of.

Reply:
We have two tests that fail without this check - I think older keys might give issues in this sense

Reply:
Maybe the tests should be fixed instead?
https://github.com/openpgpjs/openpgpjs/search?q=enc.publicKeyAlgorithm+%3D+%27rsa_encrypt%27